client/app/scripts/superdesk-packaging/packaging.js

```diff
@@ -91,11 +92,22 @@
                 origGroups.push(targetGroup);
             }
             _.each(items, function(item) {
-                targetGroup.refs.push(getReferenceFor(item));
+                targetGroup.refs.push(self.getReferenceFor(item));
             });
             _.extend(current, {groups: origGroups});
         };
 
+        this.isAdded = function(pkg, item) {
+            for (var i = 0; i < pkg.groups.length; i++) {
+                for (var j = 0; j < pkg.groups[i].refs.length; j++) {
+                    if (pkg.groups[i].refs[j].guid === item._id) {
+                        return true;
+                    }
+                }
+            }
+            return false;
```

Comment:
Instead of manually iterating over both lists, it's IMO better to simply use the built-in Array methods, making the code more concise and less error prone.

```js
return pkg.groups.some(function (group) {
    return group.refs.some(function (ref) {
        return ref.guid === item._id;
    });
});
```